File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example/h10-w/chassis/chassis_max_vel.cpp
// control_power.cpp

#include <iostream>
#include <iomanip>
#include <thread>
#include <chrono>
#include "h10-w/h10w_chassis.h"
#include "common/common_def.h"
#include "common/LogWrapper.h"

using namespace humanoid;
using namespace humanoid::h10w;

int main() {
  LOG_INFO("=== 底盘获取设置最大速度 ===");
  H10wChassis chassis;
  float linear_velocity, angular_velocity;
  int ret = chassis.getChassisMaxVel(linear_velocity, angular_velocity);
  if (ret < 0) {
    LOG_ERROR("Get chassis max velocity failed.")
    return 0;
  } else {
    LOG_INFO(
        "Get chassis max velocity successed. linear velocity:%f, angular "
        "velocity:%f.",
        linear_velocity, angular_velocity)
  }
  ret = chassis.setChassisMaxVel(linear_velocity, angular_velocity);
  if (ret < 0) {
    LOG_ERROR("Set chassis max velocity failed.")
    return 0;
  } else {
    LOG_INFO("Set chassis max velocity successed.")
  }

  return 0;
}

File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example_py/h10-w/chassis/chassis_max_vel.py
import humanoid_sdk_py
import humanoid_sdk_py.h10w as h10w
import time

print("=== 底盘获取设置最大速度 ===")
h10w_chassis = h10w.H10wChassis()
time.sleep(1)
linear_velocity = 0.0
angular_velocity = 0.0
ret, (linear_velocity, angular_velocity) = h10w_chassis.getChassisMaxVel()
if ret != 0:
    print("Get chassis max velocity failed.")
    exit(-1)

print(
    "Get chassis max velocity successed. linear velocity:%f, angular velocity:%f."
    % (linear_velocity, angular_velocity)
)

ret = h10w_chassis.setChassisMaxVel(linear_velocity, angular_velocity)
if ret != 0:
    print("Set chassis max velocity failed.")
    exit(-1)

print("Set chassis max velocity successed.")
